contribute/design/1.x/cstor-operator/doc.md

```diff
@@ -46,107 +46,154 @@ Following is the proposed CSPC schema in go struct:
+
+	// Tolerations, if specified, the pool pod's tolerations.
+	Tolerations []corev1.Toleration `json:"tolerations"`
+
+	// PriorityClassName if specified applies to this pool pod
+	// If left empty, DefaultPriorityClassName is applied.
+	// (See CStorPoolClusterSpec.DefaultPriorityClassName)
+	// If both are empty, not priority class is applied.
+	PriorityClassName string `json:"priorityClassName"`
 }
 
 // RaidGroup contains the details of a raid group for the pool
 type RaidGroup struct {
-  // Type is the raid group type
-  // Supported values are: stripe, mirror, raidz and raidz2
-
-  // stripe -- stripe is a raid group which divides data into blocks and
-  // spreads the data blocks across multiple block devices.
-
-  // mirror -- mirror is a raid group which does redundancy
-  // across multiple block devices.
-
-  // raidz -- RAID-Z is a data/parity distribution scheme like RAID-5, but uses dynamic stripe width.
-  // radiz2 -- TODO
-  // Optional -- defaults to `defaultRaidGroupType` present in `PoolConfig`
-  Type string `json:"type"`
-  // IsWriteCache is to enable this group as a write cache.
-  IsWriteCache bool `json:"isWriteCache"`
-  // IsSpare is to declare this group as spare which will be
-  // part of the pool that can be used if some block devices
-  // fail.
-  IsSpare bool `json:"isSpare"`
-  // IsReadCache is to enable this group as read cache.
-  IsReadCache bool `json:"isReadCache"`
-  // BlockDevices contains a list of block devices that
-  // constitute this raid group.
-  BlockDevices []CStorPoolClusterBlockDevice `json:"blockDevices"`
+	BlockDevices []CStorPoolClusterBlockDevice `json:"blockDevices"`
 }
 
 // CStorPoolClusterBlockDevice contains the details of block devices that
 // constitutes a raid group.
 type CStorPoolClusterBlockDevice struct {
-  // BlockDeviceName is the name of the block device.
-  BlockDeviceName string `json:"blockDeviceName"`
-  // Capacity is the capacity of the block device.
-  // It is system generated
-  Capacity string `json:"capacity"`
-  // DevLink is the dev link for block devices
-  DevLink string `json:"devLink"`
+	// BlockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+
+	// Capacity is the capacity of the block device.
+	// It is system generated
+	Capacity string `json:"capacity"`
+	
+	// DevLink is the dev link for block devices
+	DevLink string `json:"devLink"`
 }
 
-
-// CStorPoolClusterStatus is for handling status of pool.
+// CStorPoolClusterStatus represents the latest available observations of a CSPC's current state.
 type CStorPoolClusterStatus struct {
-  Phase    string                `json:"phase"`
-  Capacity CStorPoolCapacityAttr `json:"capacity"`
+	// CurrentProvisionedInstances is the the number of CSPI present at the current state. 
+	CurrentProvisionedInstances int32 `json:"currentProvisionedInstances"`
+	
+	// RunningInstances is the number of CSPI(s) that are not healthy but in degraded,rebuilding etc mode.
```

Comment:
Running can be misleading and considered Healthy. Would it be possible to spell out the different states of the pool like DegradedInstances, RebuildingInstances and HealthyInstances ?

Reply:
My take: RunningInstances should include Healthy as well @sonasingh46 .. and HealthyInstances is available in spec

Reply:
As discussed -- the states that will be present are : 
`ProvisionedInstances`
`DesiredInstances`
`HealthyInstances`

Reply:
Done